Formula/o/openbao.rb

```diff
@@ -0,0 +1,42 @@
+class Openbao < Formula
+  desc "Provides a software solution to manage, store, and distribute sensitive data"
+  homepage "https://openbao.org/"
+  url "https://github.com/openbao/openbao.git",
+      tag:      "v2.0.0-beta20240618",
+      revision: "1c194a17153afe958e1fc74f17e59f16b8d6d95e"
+  license "MPL-2.0"
+  head "https://github.com/openbao/openbao.git", branch: "main"
+
+  depends_on "go" => :build
+  depends_on "node" => :build
+  depends_on "yarn" => :build
+
+  uses_from_macos "curl" => :test
+
+  def install
+    ENV.prepend_path "PATH", Formula["node"].opt_libexec/"bin" # for npm
+    system "make", "bootstrap", "static-dist", "dev-ui"
```

Comment:
Does this compile with static libraries? Because it shouldn't.

Reply:
Not sure what do you mean. This is how the vault package was made, and seeing as openbao is basically the FOSS fork of vault I used the same.

Do you think it should compile with dynamic libs?

Reply:
yeah, it is the same thing in the vault